src/vs/vscode.proposed.d.ts

```diff
@@ -23,6 +23,37 @@ declare module 'vscode' {
 	export namespace window {
 		export function showOpenDialog(options: OpenDialogOptions): Thenable<Uri[]>;
 		export function showSaveDialog(options: SaveDialogOptions): Thenable<Uri>;
+
+		/**
+		 * Shows a selection list of [workspace folders](#workspace.workspaceFolders) to pick from.
+		 * Returns `undefined` if no folder is open.
+		 *
+		 * @param options Configures the behavior of the workspace folder list.
+		 * @param token A token that can be used to signal cancellation.
+		 * @return A promise that resolves to the workspace folder or `undefined`.
+		 */
+		export function showWorkspaceFolderPick(options?: WorkspaceFolderPickOptions, token?: CancellationToken): Thenable<WorkspaceFolder | undefined>;
+	}
+
+	/**
+	 * Options to configure the behaviour of the [workspace folder](#WorkspaceFolder) pick UI.
+	 */
+	export interface WorkspaceFolderPickOptions {
+
+		/**
+		 * An optional string to show as place holder in the input box to guide the user what to pick on.
+		 */
+		placeHolder?: string;
+
+		/**
+		 * Set to `true` to keep the picker open when focus moves to another part of the editor or to another window.
+		 */
+		ignoreFocusOut?: boolean;
+
+		/**
+		 * An optional function that is invoked whenever a [workspace folder](#WorkspaceFolder) is selected.
+		 */
+		onDidSelectWorkspaceFolder?(folder: WorkspaceFolder): any;
```

Comment:
Questionable, maybe wait for a consumer...

Reply:
Fixed via https://github.com/Microsoft/vscode/pull/34431/commits/38511f556299345d51986711a0fe6041178b07dd